lib/charms/kubernetes_charm_libraries/v0/multus.py

```diff
@@ -178,8 +178,10 @@ def pod_is_ready(
         """
         try:
             pod = self.client.get(Pod, name=pod_name, namespace=self.namespace)
-        except ApiError:
-            raise KubernetesMultusError(f"Pod {pod_name} not found")
+        except ApiError as e:
+            if e.status.reason != "Unauthorized":
+                raise KubernetesMultusError(f"Pod {pod_name} not found")
+            return False
```

Comment:
Would it make sense to still log the error for easier debugging in the future?

Reply:
Yeah, I actually thought about it. Resigned, because the original didn't have it.
Added now.